File: domain/src/main/java/com/aceplus/domain/model/posm/PosmByCustomer.java
package com.aceplus.domain.model.posm;

/**
 * Created by yma on 2/21/17.
 *
 * PosmByCustomer
 */

public class PosmByCustomer {

    /**
     * invoiceNo
     */
    private String invoiceNo;

    /**
     * invoiceDate
     */
    private String invoiceDate;

    /**
     * customerId
     */
    private int customerId;

    /**
     * stockId
     */
    private int stockId;

    /**
     * shopTypeId
     */
    private int shopTypeId;

    /**
     * saleManId
     */
    private String saleManId;

    /**
     * quantity
     */
    private int quantity;

    /**
     * price
     */
    private double price;

    /**
     * Getter method for invoiceNo.
     *
     * @return invoiceNo
     */
    public String getInvoiceNo() {
        return invoiceNo;
    }

    /**
     * Setter method for invoiceNo
     *
     * @param invoiceNo invoice number
     */
    public void setInvoiceNo(String invoiceNo) {
        this.invoiceNo = invoiceNo;
    }

    /**
     * Getter method for invoiceDate.
     *
     * @return invoiceDate
     */
    public String getInvoiceDate() {
        return invoiceDate;
    }

    /**
     * Setter method for invoiceDate
     *
     * @param invoiceDate invoice date
     */
    public void setInvoiceDate(String invoiceDate) {
        this.invoiceDate = invoiceDate;
    }

    /**
     * Getter method for customerId.
     *
     * @return customerId
     */
    public int getCustomerId() {
        return customerId;
    }

    /**
     * Setter method for customerId
     *
     * @param customerId customer id
     */
    public void setCustomerId(int customerId) {
        this.customerId = customerId;
    }

    /**
     * Getter method for stockId.
     *
     * @return stockId
     */
    public int getStockId() {
        return stockId;
    }

    /**
     * Setter method for stockId
     *
     * @param stockId stock id number
     */
    public void setStockId(int stockId) {
        this.stockId = stockId;
    }

    /**
     * Getter method for shopTypeId.
     *
     * @return shopTypeId
     */
    public int getShopTypeId() {
        return shopTypeId;
    }

    /**
     * Setter method for shopTypeId
     *
     * @param shopTypeId shop type id number
     */
    public void setShopTypeId(int shopTypeId) {
        this.shopTypeId = shopTypeId;
    }

    /**
     * Getter method for saleManId.
     *
     * @return saleManId
     */
    public String getSaleManId() {
        return saleManId;
    }

    /**
     * Setter method for saleManId
     *
     * @param saleManId sale man id number
     */
    public void setSaleManId(String saleManId) {
        this.saleManId = saleManId;
    }

    /**
     * Getter method for quantity.
     *
     * @return quantity
     */
    public int getQuantity() {
        return quantity;
    }

    /**
     * Setter method for quantity
     *
     * @param quantity quantity of product
     */
    public void setQuantity(int quantity) {
        this.quantity = quantity;
    }

    /**
     * Getter method for price.
     *
     * @return price
     */
    public double getPrice() {
        return price;
    }

    /**
     * Setter method for price
     *
     * @param price product price
     */
    public void setPrice(double price) {
        this.price = price;
    }
}

File: domain/src/main/java/com/aceplus/domain/entity/posm/POSMByCustomer.kt
package com.aceplus.domain.entity.posm

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName


@Entity(tableName = "posm_by_customer")
class POSMByCustomer {

    @ColumnInfo(name = "invoice_no")
    @SerializedName("invoice_no")
    @Expose
    var invoice_no: String? = null

    @ColumnInfo(name = "invoice_date")
    @SerializedName("invoice_date")
    @Expose
    var invoice_date: String? = null

    @ColumnInfo(name = "customer_id")
    @SerializedName("customer_id")
    @Expose
    var customer_id: Int = 0

    @ColumnInfo(name = "stock_id")
    @SerializedName("stock_id")
    @Expose
    var stock_id: Int = 0

    @ColumnInfo(name = "shop_type_id")
    @SerializedName("shop_type_id")
    @Expose
    var shop_type_id: Int = 0

    @ColumnInfo(name = "sale_man_id")
    @SerializedName("sale_man_id")
    @Expose
    var sale_man_id: Int = 0

    @ColumnInfo(name = "quantity")
    @SerializedName("quantity")
    @Expose
    var quantity: Double = 0.0

    @ColumnInfo(name = "price")
    @SerializedName("price")
    @Expose
    var price: Double = 0.0

    @ColumnInfo(name = "delete_flag")
    @SerializedName("delete_flag")
    @Expose
    var delete_flag: String? = null

}